js/core/base-runner.js

```diff
@@ -70,7 +70,7 @@ if (window.console) {
 
 
 define(
-    [],
+    [ "Promise" ],
```

Comment:
I think the problem is that `profile-w3c-common.js` needs to define that it needs Promise, not particularly this module (as it's any module that can use it, and because Promise is attached to the global scope). 

I'm going to review the RequireJS docs about what hackery the Optimizer does, and why it's not being packaged properly... but I suspect it's just the above. 

Will report back.